sdk/testing/stepwise/environments/docker/environment.go

```diff
@@ -0,0 +1,851 @@
+	//clean up networks
+	if rc.networkID != "" {
+		cli, err := docker.NewClientWithOpts(docker.FromEnv, docker.WithVersion("1.40"))
+		if err != nil {
+			return err
+		}
+		if err := cli.NetworkRemove(context.Background(), rc.networkID); err != nil {
+			return err
+		}
+	}
+
+	return result
+}
+
+// MountPath returns the path that the plugin under test is mounted at. If a
+// MountPathPrefix was given, the mount path uses the prefix with a uuid
+// appended. The default is the given PluginName with a uuid suffix.
+func (dc *DockerCluster) MountPath() string {
+	if dc.mountPath != "" {
+		return dc.mountPath
+	}
+
+	uuidStr, _ := uuid.GenerateUUID()
+	prefix := dc.PluginName
+	if dc.MountOptions.MountPathPrefix != "" {
+		prefix = dc.MountOptions.MountPathPrefix
+	}
+
+	dc.mountPath = fmt.Sprintf("%s_%s", prefix, uuidStr)
+	if dc.MountOptions.PluginType == stepwise.PluginTypeCredential {
+		dc.mountPath = fmt.Sprintf("%s/%s", "auth", dc.mountPath)
+	}
+
+	return dc.mountPath
+}
+
+// RootToken returns the root token of the cluster, if set
+func (dc *DockerCluster) RootToken() string {
+	return dc.rootToken
+}
+
+// Name returns the name of this environment
+func (dc *DockerCluster) Name() string {
+	return "Docker"
+}
+
+// Client returns a clone of the configured Vault API client.
+func (dc *DockerCluster) Client() (*api.Client, error) {
+	if len(dc.ClusterNodes) > 0 {
+		if dc.ClusterNodes[0].Client != nil {
+			c, err := dc.ClusterNodes[0].Client.Clone()
+			if err != nil {
+				return nil, err
+			}
+			c.SetToken(dc.ClusterNodes[0].Client.Token())
+			return c, nil
+		}
+	}
+
+	return nil, errors.New("no configured client found")
+}
+
+func (n *DockerClusterNode) Name() string {
+	return n.Cluster.ClusterName + "-" + n.NodeID
+}
+
+func (rc *DockerCluster) Initialize(ctx context.Context) error {
+	client, err := rc.ClusterNodes[0].CreateAPIClient()
+	if err != nil {
+		return err
+	}
+
+	var resp *api.InitResponse
+	for ctx.Err() == nil {
+		resp, err = client.Sys().Init(&api.InitRequest{
+			SecretShares:    3,
+			SecretThreshold: 3,
+		})
+		if err == nil && resp != nil {
+			break
+		}
+		time.Sleep(500 * time.Millisecond)
+	}
+	if err != nil {
+		return err
+	}
+	if resp == nil {
+		return fmt.Errorf("nil response to init request")
+	}
+
+	for _, k := range resp.Keys {
+		raw, err := hex.DecodeString(k)
+		if err != nil {
+			return err
+		}
+		rc.barrierKeys = append(rc.barrierKeys, raw)
+	}
+
+	for _, k := range resp.RecoveryKeys {
+		raw, err := hex.DecodeString(k)
+		if err != nil {
+			return err
+		}
+		rc.recoveryKeys = append(rc.recoveryKeys, raw)
+	}
+
+	rc.rootToken = resp.RootToken
+
+	ctx, cancel := context.WithTimeout(ctx, 15*time.Second)
+	defer cancel()
+
+	// Unseal
+	for j, node := range rc.ClusterNodes {
+		// copy the index value, so we're not reusing it in deeper scopes
+		i := j
+		client, err := node.CreateAPIClient()
+		if err != nil {
+			return err
+		}
+		node.Client = client
+
+		if i > 0 && rc.RaftStorage {
```

Comment:
With the addition of Raft for HA Storage (`ha_storage` coming in 1.5), this step needs to happen after unsealing. So there probably needs to be another bool (or change the existing bool to a tri-state value) to support this.

Reply:
Deferring and tracking this internally, to follow up in the coming weeks